### PR TITLE
fix: edit translated caroussel data

### PR DESCRIPTION
**Pull request by EFF**
Notice this line is the original code, the comment mentions we should ignore array arrays of empty objects
https://github.com/botpress/studio/blob/fed7ac66da61b6ac3a5cc0d43ca1106806934e7c/packages/studio-ui/src/web/util/NodeFormData.ts#L18-L34

In this code at line 27, we return undefined if the array has **not empty** objects `!formData[x].filter(_.isEmpty).length`

I rewrote the logic for something much simpler, yet readable.

Fixes https://github.com/botpress/studio/issues/108
Closes : DEV-1825 & DEV-1086

### CODE REVIEW

**Comment by laurentlp** [comment]
@EFF What's the link between this PR and DEV-1580?

File: packages/studio-ui/src/web/util/NodeFormData.ts [hunk 1]
```diff
@@ -1,36 +1,24 @@
 import _ from 'lodash'
 
 export const getFormData = (node: any, contentLang: string, defaultLanguage: string, defaultValue: any = {}): any => {
```
Comment (laurentlp) [inline packages/studio-ui/src/web/util/NodeFormData.ts:3]: Can you add missing typings!

**Review by laurentlp — COMMENTED** [review]
Small comment other than that LGTM!

**Review by laurentlp — DISMISSED** [review]
Tested and working fine! Good job!!

**Comment by EFF** [comment]
@laurentlp I added typings but it's the worst typing we could add as by debugging, there is a single key in the object and its any. We also read a second key that is also a type any and return it as is if it's defined (although I didn't manage to see it defined).

To be honest that whole file should be re-written properly

**Review by laurentlp — APPROVED** [review]

**Pull request merged**